drivers/ioexpander/gpio.c

```diff
@@ -347,8 +349,9 @@ static int gpio_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
           FAR bool *ptr = (FAR bool *)((uintptr_t)arg);
           DEBUGASSERT(ptr != NULL);
 
+          DEBUGASSERT(dev->gp_ops->go_read != NULL);
           ret = dev->gp_ops->go_read(dev, ptr);
-          DEBUGASSERT(ret < 0 || *ptr == 0 || *ptr == 1);
```

Comment:
why remove ret < 0?

Reply:
Please correct me if my thought process is wrong, but I believe ret should always be zero when things are going right. ret is smaller than zero when go_read returned an error code. So this debugassert will always trigger unless something goes actually wrong. So it was wrong in the first place.

Reply:
if ret < 0, we should skip check *ptr equal 0 or 1 since the check is meaningless in case of fail.

Reply:
Ah, I think I get it now! Thanks. I have changed it back.